src/native/corehost/hostpolicy/deps_resolver.cpp

```diff
@@ -735,17 +735,23 @@ void deps_resolver_t::resolve_additional_deps(const arguments_t& args, const dep
 
     for (pal::string_t json_file : m_additional_deps_files)
     {
-        m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
-            new deps_json_t(true, json_file, rid_fallback_graph)));
+        if (rid_fallback_graph != nullptr)
+        {
+            m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
```

Comment:
This seems to be the only place where we call these overloads of the `deps_json_t` .ctor. Would it make sense to move the if into the .ctor - so that it accepts a pointer and can handle `nullptr` internally. It seems weird to have two overloads just for this if.